CONFIGURATION.md

```diff
@@ -0,0 +1,54 @@
+<!-- vim: filetype=markdown colorcolumn=80
+-->
+# helm-charts-flex Configuration
+
+[![Apache 2.0 License](https://img.shields.io/github/license/spiffe/helm-charts)](https://opensource.org/licenses/Apache-2.0)
+[![Development Phase](https://github.com/spiffe/spiffe/blob/main/.img/maturity/dev.svg)](https://github.com/spiffe/spiffe/blob/main/MATURITY.md#development)
+[![Artifact Hub](https://img.shields.io/endpoint?url=https://artifacthub.io/badge/repository/spiffe)](https://artifacthub.io/packages/search?repo=spiffe)
+
+## Agent Configuration
+
+### Agent Health Checks
+
+The agent can expose additional endpoint that can be used for health checking.
+It is enabled by default, but can be disabled by setting **.agent.healthCheck.enable**
+to *false*.
+
+The values associated with the agent health checks include
+
+| Path                          | Type   | Default     | 
+| ----------------------------- | ------ | ----------- |
+| agent.healthCheck.enable      | bool   | true        |
+| agent.healthCheck.bindAddress | string | "localhost" |
+| agent.healthCheck.bindPort    | int    | 80          |
```

Comment:
Not sure these are correct...
If the pod is not running net=host, then localhost will be relative to the pod, and be unreachable by kubelet for testing I think?

If the pod is running net=host, then port 80 is squatting on a very common port used for other things and a different port would be better to choose as a default.

I don't know if spire-agent can run without net=host and function properly.

Reply:
They are correct, in the sense that the default values are those of the spire-agent.

Whether these items have any purpose in a K8s deployment is unknown.  It will need testing when we have a DeploymentSet that actually deploys Pods.  Odds are that only localhost, 127.0.0.1, and ::1 bindAddresses will make sense in a K8s context, and it is unclear at the moment if they need nodePort (or other support items) in the DaemonSet, or if this approach is even workable.

If it is not, we can always change the default of .agent.healthCheck.enabled to false and adjust the unit tests.

For now, let's accept the code as-is until we can see if localhost, 127.0.0.1, or ::1 works in a running DaemonSet.  I don't think we need to schedule a "reminder" Issue, as it will be broken if the approach is not suitable.

Reply:
Added values.schema.json to enforce only bindAddress values of localhost, 127.0.0.1, and ::1
As the DaemonSet can't have a single IP address apply to each of the pod instances, this is likely the only approach possible.  
Even with these settings, we might need to rework the solution if the localhost / aliases don't work
Let's consider this done till we have a functional deployment